blas/testblas/level1double.go

```diff
@@ -1731,461 +1732,466 @@ func DrotgTest(t *testing.T, d Drotger, skipExtreme bool) {
+			pWant: blas.DrotmParams{
+				Flag: blas.Rescaling,
+				H:    [4]float64{1, -0.0004096, 1000, 4096},
+			},
+		},
+		{
+			name:   "D1_Med_D2_Big_Flag_1",
+			d1:     20000000000,
+			d2:     0.4,
+			x1:     80000000,
+			y1:     80000000000000000,
+			d1Want: 0.39999998000000103,
+			d2Want: 1192.092835903171,
+			x1Want: 8.0000004e+16,
+			pWant: blas.DrotmParams{
+				Flag: blas.Rescaling,
+				H:    [4]float64{50, -4096, 1, 4.096e-06},
+			},
+		},
+		{
+			name:   "RD1_Med_RD2_Small_Flag_0",
+			d1:     1.2,
+			d2:     0.000000000045,
+			x1:     2.7,
+			y1:     8,
+			d1Want: 1.1999999996049382,
+			d2Want: 0.0007549747197514486,
+			x1Want: 2.700000000888889,
+			pWant: blas.DrotmParams{
+				Flag: blas.Rescaling,
+				H:    [4]float64{1, -0.0007233796296296296, 1.1111111111111111e-10, 0.000244140625},
+			},
+		},
+		{
+			name:   "RD1_Med_RD2_Small_Flag_1",
+			d1:     1.2,
+			d2:     0.000000000045,
+			x1:     2.7,
+			y1:     80000000000,
+			d1Want: 0.0007549747199770676,
+			d2Want: 1.19999999996355,
+			x1Want: 1.9531250000593264e+07,
+			pWant: blas.DrotmParams{
+				Flag: blas.Rescaling,
+				H:    [4]float64{0.0002197265625, -1, 0.000244140625, 3.375e-11},
+			},
+		},
+		{
+			name:   "D1_Small_D2_Big_Flag_1",
+			d1:     120000000000000000,
+			d2:     0.000000000012345,
+			x1:     0.08,
+			y1:     8000000000000,
+			d1Want: 0.00010502490698765249,
+			d2Want: 216.1836123957717,
+			x1Want: 3.8516669198055897e+09,
+			pWant: blas.DrotmParams{
+				Flag: blas.Rescaling,
+				H:    [4]float64{2.3731773997569866e+10, -1.6777216e+07, 0.000244140625, 1.6777216e-07},
+			},
+		},
+		{
+			name:   "RD1_Small_RD2_Med_Flag_0",
+			d1:     0.0000000002,
+			d2:     20,
+			x1:     0.8,
+			y1:     0.000000008,
+			d1Want: 0.003355409645903541,
+			d2Want: 19.99980000199998,
+			x1Want: 0.000195314453125,
+			pWant: blas.DrotmParams{
+				Flag: blas.Rescaling,
+				H:    [4]float64{0.000244140625, -1e-08, 0.24414062499999997, 1},
+			},
+		},
+		{
+			name:   "RD1_Small_RD2_Med_Flag_1",
+			d1:     0.02,
+			d2:     0.000000000004,
+			x1:     0.008,
+			y1:     8000000,
+			d1Want: 6.710886366445568e-05,
+			d2Want: 0.019999999900000003,
+			x1Want: 1953.125009765625,
+			pWant: blas.DrotmParams{
+				Flag: blas.Rescaling,
+				H:    [4]float64{0.0012207031250000002, -1, 0.000244140625, 1e-09},
+			},
+		},
+		{
+			// Values consistent with the low precision output posted at the  OpenBLAS issue.
+			// See https://github.com/xianyi/OpenBLAS/issues/1452.
+			name:   "OpenBLAS#1452",
+			d1:     5.9e-8,
+			d2:     5.960464e-8,
+			x1:     1,
+			y1:     150,
+			d1Want: 0.9999559282289687,
+			d2Want: 0.9898121986058326,
+			x1Want: 0.03662270484346241,
+			pWant: blas.DrotmParams{
+				Flag: blas.Rescaling,
+				H:    [4]float64{1.6110934624105326e-06, -0.000244140625, 0.000244140625, 1.6276041666666668e-06},
+			},
+		},
+
+		// Test cases with values taken from BLAS/TESTING/dblat1.f in the reference.
```

Comment:
It's not visible in the diff: I modified these test cases to use "exact" values from the reference.

Reply:
Why were we previously not using the exact values?

Reply:
The test cases were added in https://github.com/gonum/gonum/pull/420. It looks like you used the runtime output of a printf statement, so they're practically the same values only in a different representation.

Reply:
Yes. That seems likely. I have no memory of that.

I've gone through the netlib/BLAS tests and I'm convinced that that is what I did.